ocs_ci/helpers/mcg_stress_helper.py

```diff
@@ -0,0 +1,618 @@
+                f"DOWNLOAD OP: Successfully completed object download "
+                f"operation on bucket {bucket_name} under prefix {prev_iteration}"
+            )
+            break
+
+
+def delete_objects_in_batches(bucket, batch_size):
+    """
+    Delete objects from the bucket in batches
+
+    Args:
+        bucket (tuple): Tuple consisting of backend storage type and bucket object
+        batch_size (int): Number of objects to delete at a time
+
+    """
+    bucket_type, bucket_obj = bucket
+    bucket_name = bucket_obj.name
+    if bucket_type.upper() == "RGW":
+        mcg_obj = OBC(bucket_name)
+    else:
+        mcg_obj = MCG()
+
+    logger.info(f"Deleting objects in bucket {bucket_name} in batches of {batch_size}")
+    total_objs_deleted = 0
+    for obj_batch in list_objects_in_batches(
+        mcg_obj, bucket_name, batch_size=batch_size, yield_individual=False
+    ):
+        s3_delete_objects(mcg_obj, bucket_name, obj_batch)
+        total_objs_deleted += batch_size
+        logger.info(
+            f"Total objects deleted {total_objs_deleted} in bucket {bucket_name}"
+        )
+
+
+def run_background_cluster_checks(scale_noobaa_db_pv, event=None, threading_lock=None):
+    """
+    Run background checks to verify noobaa health
+    and cluster health overall
+
+        1. Check Noobaa Health
+        2. Check Ceph Health
+        3. Check Noobaa db usage
+        4. Check for any alerts
+        5. Memory and CPU utilization
+
+    """
+    ceph_cluster = CephCluster()
+    prometheus_api = PrometheusAPI(threading_lock=threading_lock)
+
+    logger.info(
+        "\n"
+        "\nNow starting background check operations to check the following"
+        "\n1. Nooba Health"
+        "\n2. Ceph Health"
+        "\n3. Noobaa DB usage"
+        "\n4. Prometheus Alerts"
+        "\n5. Memory and CPU utilization for Noobaa pods"
+        "\n"
+    )
+
+    @retry(NoobaaHealthException, tries=10, delay=60)
+    def check_noobaa_health():
+
+        while True:
+
+            ceph_cluster.noobaa_health_check()
+            logger.info(
+                "\n"
+                "\n[BACKGROUND CHECK]"
+                "\nNoobaa is healthy... rechecking in 5 minute"
+                "\n"
+            )
+
+            if event.is_set():
+                logger.info("[BACKGROUND CHECK] Stopping the Noobaa health check")
+                break
+
+            time.sleep(300)
+
+    @retry(CephHealthException, tries=10, delay=60)
+    def check_ceph_health():
+
+        while True:
+
+            if ceph_cluster.get_ceph_health() == constants.CEPH_HEALTH_ERROR:
+                raise CephHealthException
+            logger.info(
+                "\n"
+                "\n[BACKGROUND CHECK]"
+                "\nCeph is healthy... rechecking in 5 minute"
+                "\n"
+            )
+
+            if event.is_set():
+                logger.info("[BACKGROUND CHECK] Stopping the Ceph health check")
+                break
```

Comment:
You are performing the checks in each individual function. It might be a good idea to consolidate them into a single mechanism which would be more efficient and easier to manage